associated-token-account/program/src/processor.rs

```diff
@@ -1,23 +1,45 @@
 //! Program state processor
 
 use crate::*;
+use crate::{instruction::AssociatedTokenAccountInstruction, tools::create_pda_account};
+use borsh::BorshDeserialize;
 use solana_program::{
     account_info::{next_account_info, AccountInfo},
     entrypoint::ProgramResult,
     msg,
-    program::{invoke, invoke_signed},
+    program::invoke,
     program_error::ProgramError,
     pubkey::Pubkey,
     rent::Rent,
-    system_instruction,
     sysvar::Sysvar,
 };
 
 /// Instruction processor
 pub fn process_instruction(
     program_id: &Pubkey,
     accounts: &[AccountInfo],
-    _input: &[u8],
+    input: &[u8],
+) -> ProgramResult {
+    let instruction = if input.is_empty() {
+        AssociatedTokenAccountInstruction::CreateAssociatedTokenAccount
+    } else {
+        AssociatedTokenAccountInstruction::try_from_slice(input)
+            .map_err(|_| ProgramError::InvalidInstructionData)?
+    };
+
+    msg!("ASSOCIATED-TOKEN-ACCOUNT-INSTRUCTION: {:?}", instruction);
```

Comment:
No need to yell, this is sufficient since log output is already clearly scoped by program id
```suggestion
    msg!("{:?}", instruction);
```

Reply:
I actually prefer to have the instruction prefixed by program name.
I've recently spent a substantial amount of time looking at logs of inspected transactions and it's quite hard because you have to memorise program ids and they don't stand out in the logs with all the other Pubkeys

Reply:
The ATA program id starts with `AToken..` so it's pretty clear what it is :)

Reply:
Yeah, it's clear only if you already know that :)

Reply:
I feel pretty strongly about not adding this yelly log message.

Reply:
removed